### PR TITLE
Update panels

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

File: src/_locales/en/messages.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "message": "Facebook Container isolates your Facebook activity from the rest of your web activity in order to prevent Facebook from tracking you outside of the Facebook website via third party cookies.",
         "description": "Description of the extension. DO NOT TRANSLATE \"Facebook Container\"."
     },
-    "onFacebookHeader": {
+    "on-facebook-subhead": {
```
Comment (groovecoder) [inline src/_locales/en/messages.json]: IIRC we don't like to change message strings once they've been published to Pontoon. These should probably be moved to whole new message IDs.

File: src/background.js [hunk 2]
```diff
@@ -384,8 +386,9 @@ async function updateBrowserActionIcon (tab) {
       });
       browser.browserAction.setBadgeText({tabId: tab.id, text: " "});
     }
-  } else {
-    browser.browserAction.setPopup({tabId: tab.id, popup: "./panel2.html"});
+  } else { 
+    browser.storage.local.set({"CURRENT_PANEL": "no-trackers"});
```
Comment (groovecoder) [inline src/background.js]: This is what's setting the panel back to the "no-trackers" state when it's re-opened. Filed https://github.com/mozilla/contain-facebook/issues/270

File: src/panel.js [hunk 3]
```diff
@@ -1,8 +1,152 @@
...
+const clearPanel = (wrapper) => {
+  while (wrapper.firstChild) {
+    wrapper.removeChild(wrapper.firstChild);
+  }
+};
+
+
+// adds "Facebook Container" to top of all panels
+const addHeader = (wrapper) => {
+  let el = document.createElement("H1");
```
Comment (groovecoder) [inline src/panel.js]: nit: lowercase the element names?

File: src/panel.js [hunk 4]
```diff
@@ -15,17 +159,146 @@ document.addEventListener("DOMContentLoaded", async () => {
...
+    addParagraph(contentWrapper, `${panelId}-p2`);
+  }
+
+  if (panelId !== "on-facebook") {
+    let link = document.createElement("A");
+    link["id"] = "learn-more";
+    link.classList.add("open-sumo");
+    link["href"] = "https://support.mozilla.org"; // need Facebook Container SUMO url. // need UTM params? // open in new or same window?
+    setClassAndAppend(contentWrapper, link);
+    link.addEventListener("click", () => window.close());
```
Comment (groovecoder) [inline src/panel.js]: This could be refactored into a `createLearnMoreLink()` function?

File: src/panel.js [hunk 5]
```diff
@@ -15,17 +159,146 @@ document.addEventListener("DOMContentLoaded", async () => {
...
+  }
+
+  // add div.fw-bottom-btn (full-width button at the bottom of the panels).
+  contentWrapper = addDiv(pageWrapper, "fw-bottom-btn");
+
+  let button = document.createElement("BUTTON");
+  button.classList.add("highlight-on-hover", "open-onboarding");
+  contentWrapper.appendChild(button);
```
Comment (groovecoder) [inline src/panel.js]: `createOnboardingButton` function?

File: src/panel.js [hunk 6]
```diff
@@ -15,17 +159,146 @@ document.addEventListener("DOMContentLoaded", async () => {
...
+
+  contentWrapper = button;
+
+  // add span#how-fbc-works and arrow icon
+  let span = document.createElement("SPAN");
+  span["id"] = "how-fbc-works";
+  setClassAndAppend(contentWrapper, span);
```
Comment (groovecoder) [inline src/panel.js]: `createHowFBCWorksSpan`?

File: src/panel.js [hunk 7]
```diff
@@ -15,17 +159,146 @@ document.addEventListener("DOMContentLoaded", async () => {
...
+  let stringId = `onboarding${panelId}`;
+
+  // clear out existing panel elements (if there are any)
+  clearPanel(page);
+
+  let el = addHeader(pageWrapper);
+
+  el = document.createElement("BUTTON");
+  el.classList.add("btn-return", "arrow-left");
+  pageWrapper.appendChild(el);
```
Comment (groovecoder) [inline src/panel.js]: `buildReturnButton`?

File: src/panel.css [hunk 8]
```diff
@@ -1,51 +1,248 @@
...
+}
+
+.bottom-btns { /* Bottom "Cancel, Back, Next, Done" buttons - onboarding panels only */
+  width: 100%;
+}
+
+.bottom-btn {
+  border-bottom: 0;
+  line-height: 103%;
+  height: 3.37rem;
```
Comment (maxxcrawford) [inline src/panel.css]: Recommend changing this a static pixel height of `45px`. It should resolve the additional class of `.onboarding3 .bottom-btn`.

Comment (maxxcrawford) [inline src/panel.css]: Recommend changing this a static pixel height of `45px`. It should resolve the additional class of `.onboarding3 .bottom-btn`.

File: src/panel.css [hunk 9]
```diff
@@ -1,51 +1,248 @@
...
+  background-repeat: no-repeat;
+  background-size: contain;
+}
+
+/* Bottom button hacks until a more reasonable solution presents itself */
+/* This eliminates (The-Powers-That-Be-Willing) a strange little white gap between
+   the bottom edge of the 'bottom-btns' and the bottom edge of the panel. */
+
+.onboarding3 .bottom-btn {
+  height: 3.41rem;
```
Comment (maxxcrawford) [inline src/panel.css]: I recommend rewriting this attribute to use a static `px` unit instead of `rem`.

File: src/background.js [hunk 10]
```diff
@@ -468,7 +466,6 @@ async function blockFacebookSubResources (requestDetails) {
     // See https://github.com/mozilla/contain-facebook/issues/23
     // Sometimes this add-on is installed but doesn't get a facebookCookieStoreId ?
     // eslint-disable-next-line no-console
-    console.log(error);
```
Comment (groovecoder) [inline src/background.js]: We can use `console.error` here - our `eslint` rules allow it and it helps surface the error message.

File: src/panel.js [hunk 11]
```diff
@@ -190,115 +216,81 @@ document.addEventListener("DOMContentLoaded", async () => {
...
 
 
 const buildOnboardingPanel = (panelId) => {
-  console.log(`building onboarding panel no. ${panelId} `);
   let page = document.body;
-  // let pageWrapper = document.body;
-  let pageWrapper = document.createDocumentFragment();
+  clearPanel(page);
 
+  let fragment = document.createDocumentFragment();
```
Comment (groovecoder) [inline src/panel.js]: prefer `const` over `let`

File: src/panel.js [hunk 12]
```diff
@@ -190,115 +216,81 @@ document.addEventListener("DOMContentLoaded", async () => {
...
+
   onboardingLinks.forEach(link => {
     link.addEventListener("click", () => buildOnboardingPanel(1));
   });
 };
 
 
 const buildOnboardingPanel = (panelId) => {
-  console.log(`building onboarding panel no. ${panelId} `);
   let page = document.body;
```
Comment (groovecoder) [inline src/panel.js]: `const page =`

File: src/panel.js [hunk 13]
```diff
@@ -190,115 +216,81 @@ document.addEventListener("DOMContentLoaded", async () => {
...
-  // let pageWrapper = document.body;
-  let pageWrapper = document.createDocumentFragment();
+  clearPanel(page);
 
+  let fragment = document.createDocumentFragment();
   let stringId = `onboarding${panelId}`;
```
Comment (groovecoder) [inline src/panel.js]: `const`

**Comment by groovecoder** [comment]
Ignoring Travis failures on this merge to `2.0` - we'll fix them when we merge from `2.0` to `master`

**Pull request merged**